src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs

```diff
@@ -53,7 +48,7 @@ private static Location GetLocation(Symbol symbol)
                     }
                     else
                     {
-                        ValidateDeclarationNameConflictsInScope(tp, diagnostics);
+                        ValidateDeclarationNameConflictsInScope(tp, diagnostics, outsideContainingSymbol: true);
```

Comment:
>outsideContainingSymbol: true [](start = 81, length = 29)

It looks like there is only one call site for the ```ValidateParameterNameConflicts``` function, it feels like it would be simpler to pass a flag to suppress this call altogether rather than adding a parameter to ```ValidateDeclarationNameConflictsInScope``` and to ```ValidateNameConflictsInScope``` to deal with this special case there.